mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -1342,23 +1349,28 @@ Block *ConversionPatternRewriterImpl::applySignatureConversion(
 /// of input operands.
 Value ConversionPatternRewriterImpl::buildUnresolvedMaterialization(
     MaterializationKind kind, OpBuilder::InsertPoint ip, Location loc,
-    ValueRange inputs, Type outputType, Type originalType,
+    Value valueToMap, ValueRange inputs, Type outputType, Type originalType,
```

Comment:
Could `originalType` now be deduced from `valueToMap` if needed?

Reply:
Not in the general case. E.g., we have this in `remapValues` and `origType` and `newOperand.getType()` are different here.
```
      Value castValue = buildUnresolvedMaterialization(
          MaterializationKind::Target, computeInsertPoint(newOperand),
          operandLoc, /*valueToMap=*/newOperand, /*inputs=*/newOperand,
          /*outputType=*/desiredType, /*originalType=*/origType,
          currentTypeConverter);
```